controllers/ratelimit_controller.go

```diff
@@ -18,36 +18,185 @@ package controllers
 
 import (
 	"context"
-
+	"github.com/3scale/limitador-operator/pkg/limitador"
 	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
+	"net/url"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"strconv"
 
 	limitadorv1alpha1 "github.com/3scale/limitador-operator/api/v1alpha1"
 )
 
+const rateLimitFinalizer = "finalizer.ratelimit.limitador.3scale.net"
+
+// Assumes that there's only one Limitador per namespace. We might want to
```

Comment:
@miguelsorianod @eguzki please keep this in mind while reviewing.
I think we'll need to be able to deploy multiple limitadors per namespace, but let's leave that scenario for when the use cases are more clear.